models/work-item-link-category-repository.go

```diff
@@ -0,0 +1,203 @@
+package models
+
+import (
+	"log"
+
+	"golang.org/x/net/context"
+
+	"github.com/almighty/almighty-core/app"
+	"github.com/jinzhu/gorm"
+	satoriuuid "github.com/satori/go.uuid"
+)
+
+// NewWorkItemLinkCategoryRepository creates a work item link category repository based on gorm
+func NewWorkItemLinkCategoryRepository(db *gorm.DB) *GormWorkItemLinkCategoryRepository {
+	return &GormWorkItemLinkCategoryRepository{db}
+}
+
+// GormWorkItemLinkCategoryRepository implements WorkItemLinkCategoryRepository using gorm
+type GormWorkItemLinkCategoryRepository struct {
+	db *gorm.DB
+}
+
+// Create creates a new work item link category in the repository.
+// Returns BadParameterError, ConversionError or InternalError
+func (r *GormWorkItemLinkCategoryRepository) Create(ctx context.Context, name *string, description *string) (*app.WorkItemLinkCategory, error) {
+	if name == nil || *name == "" {
+		return nil, BadParameterError{parameter: "name", value: name}
```

Comment:
I know and I already use it in the portions not committed yet. Thank you for pointing this out to me though. This PR enhances BadParameterError a bit #461.